### ISSUE TITLE
[git town sync] git fetch --prune --tags returns "rejected: would clobber existing tag"

### ISSUE THREAD

**Issue by ruudk** [git-town/git-town#3123]
I have 1 project that uses GoReleaser to publish a nightly tag on every commit to main.

When I run `git sync` on that repository, I get this error:

```command
$ git sync

[main] git fetch --prune --tags
From https://github.com/OrgName/RepoName
 ! [rejected]        nightly    -> nightly  (would clobber existing tag)
```

What would be the best way to resolve this issue?

**Comment by kevgo** [comment]
TLDR: Make `nightly` a branch instead of a tag because `nightly` is a branch and not a single release.

Background: Git tags and Git branches are both the same thing: "labels" for a commit. The only difference between them is that tags are labels that should never change, and branches are labels that are expected to change over time.

Different types of labels are useful for different things, hence Git provides both types.

Tags are useful for releases because releases should never change. Once you release version 1.2.3 of your software, and your users download it, the version of code for this release is now written in stone. There should never be another release 1.2.3 that contains a different version of the software. That's why we use Git tags for releases. Because Git tags never change.

If both your Git workspace and my Git workspace have a tag `v1.2.3` but yours points to a different commit than mine, that's an indicator that something with release 1.2.3 has gone fundamentally wrong, and humans now need to talk to each other to resolve the processes that led to the two conflicting releases. That's why Git refuses to auto-update the tag in your case.

On the other hand, if the `nightly` branch on your workspace points to a different commit than the `nightly` branch on my workspace, that just means you added some commits to this branch. When I run `git pull`, Git will happily update the label on my workspace to point to the latest commit you have added.

Does this make sense?

To resolve this issue on your machine:
- Run `git tag -d nightly` to delete the `nightly` tag from the machine.
- Run `git sync` again. It will download the new `nightly` tag. This will work now because there isn't a `nighly` tag on your machine anymore that would conflict with the tag from the remote branch.

**Comment by ruudk** [comment]
Thanks for the detailed response. I agree that a tag should not change. I made changes to the project that was using goreleaser, and it's solved now. Thanks!
